akd_local_auditor/src/storage/s3/test.rs

```diff
@@ -116,37 +116,46 @@ async fn maybe_flush_storage(shared_config: &SdkConfig, bucket_name: &str) -> Re
     if let Some(bucket_names) = buckets.buckets() {
         log::info!("Found {} buckets in the test storage", bucket_names.len());
 
-        for bucket in bucket_names {
-            if let Some(maybe_name) = bucket.name() {
-                if maybe_name == bucket_name {
-                    log::info!("Found target bucket {}, deleting.", bucket_name);
-
-                    // From: https://docs.aws.amazon.com/sdk-for-rust/latest/dg/rust_s3_code_examples.html
-                    let objects = client.list_objects_v2().bucket(bucket_name).send().await?;
-                    let mut delete_objects: Vec<ObjectIdentifier> = vec![];
-                    for obj in objects.contents().unwrap_or_default() {
-                        let obj_id = ObjectIdentifier::builder()
-                            .set_key(Some(obj.key().unwrap().to_string()))
-                            .build();
-                        delete_objects.push(obj_id);
-                    }
-                    if !delete_objects.is_empty() {
-                        log::info!(
-                            "Bucket {} has {} objects in it which must be delete first",
-                            bucket_name,
-                            delete_objects.len()
-                        );
-                        client
-                            .delete_objects()
-                            .bucket(bucket_name)
-                            .delete(Delete::builder().set_objects(Some(delete_objects)).build())
-                            .send()
-                            .await?;
-                    }
-                    client.delete_bucket().bucket(bucket_name).send().await?;
-                    break;
+        if bucket_names
+            .iter()
+            .find(|bucket| {
+                if let Some(maybe_bucket) = bucket.name() {
+                    return maybe_bucket == bucket_name;
                 }
+                false
```

Comment:
If you want a one-liner with [`map_or_else`](https://doc.rust-lang.org/std/option/enum.Option.html#method.map_or_else): 

```rust
bucket.name().map_or_else(|| false, |n| n == bucket_name)
```

Reply:
ah missed this! Oh well :p

Reply:
Yeah, no biggie! 🤷